File: oseille/src/main/java/lobstre/oseille/Command.java
package lobstre.oseille;

import java.io.IOException;
import java.text.ParseException;
import java.util.Collection;
import java.util.List;

public interface Command {
    void accepts (List<String> arguments, Collection<String> errors, String fileName);
    void execute (String fileName, List<String> arguments) throws IOException, ParseException;
}

File: src/main/scala/lobstre/oseille/Command.scala
package lobstre.oseille

import java.io.IOException
import java.text.ParseException
import java.util.{Collection => JCollection, List => JList}

trait Command {
  def accepts(arguments: JList[String], errors: JCollection[String], fileName: String)

  @throws(classOf[IOException])
  @throws(classOf[ParseException])
  def execute(fileName: String, arguments: JList[String])
}
